File: src/services/legacyAuthService.js
const axios = require('axios');

class LegacyAuthService {
  static async authenticate(username, password) {
    try {
      // TODO: Integrar com o webservice real
      // const response = await axios.post(process.env.LEGACY_WEBSERVICE_URL, {
      //   username: username,
      //   password: password
      // });

      // Simulação por enquanto - remover quando integrar com webservice real
      await new Promise(resolve => setTimeout(resolve, 100)); // Simula chamada de rede
      
      // Lógica temporária - aceita qualquer usuário para demonstração
      if (username && password) {
        return {
          success: true,
          message: 'OK'
        };
      }
      
      return {
        success: false,
        message: 'Credenciais inválidas'
      };
    } catch (error) {
      console.error('Erro na autenticação legada:', error);
      return {
        success: false,
        message: 'Erro na autenticação'
      };
    }
  }

  static async getUserPermissions(username) {
    try {
      // TODO: Integrar com servicoV.getpermissoes(username)
      await new Promise(resolve => setTimeout(resolve, 50));
      
      // Simulação - retorna permissões baseadas no usuário
      const permissions = [];
      
      // Simula diferentes níveis de acesso
      if (username.toLowerCase().includes('admin') || username === 'alexander.valerio') {
        permissions.push('@@@@'); // Acesso total
      } else {
        permissions.push('PAINEL TI');
        permissions.push('PAINEL RH');
        permissions.push('PAINEL FINANCEIRO');
      }
      
      return permissions;
    } catch (error) {
      console.error('Erro ao buscar permissões legadas:', error);
      return [];
    }
  }
}

module.exports = LegacyAuthService;

File: src/PortalCT.Infrastructure/Services/LegacyAuthService.cs
using PortalCT.Core.DTOs;

namespace PortalCT.Infrastructure.Services
{
    public interface ILegacyAuthService
    {
        Task<LoginResponse> AuthenticateAsync(string username, string password);
        Task<List<string>> GetUserPermissionsAsync(string username);
    }

    public class LegacyAuthService : ILegacyAuthService
    {
        // Aqui você integraria com o webservice existente
        // Por enquanto, vou simular a integração
        
        public async Task<LoginResponse> AuthenticateAsync(string username, string password)
        {
            try
            {
                // TODO: Integrar com WebserviceVB.loginSoapClient
                // var servicoV = new WebserviceVB.loginSoapClient();
                // var autoriza = await servicoV.loginadAsync(username, password);
                
                // Simulação por enquanto - remover quando integrar com webservice real
                await Task.Delay(100); // Simula chamada de rede
                
                // Lógica temporária - aceita qualquer usuário para demonstração
                if (!string.IsNullOrEmpty(username) && !string.IsNullOrEmpty(password))
                {
                    return new LoginResponse
                    {
                        IsSuccess = true,
                        Message = "OK"
                    };
                }
                
                return new LoginResponse
                {
                    IsSuccess = false,
                    Message = "Credenciais inválidas"
                };
            }
            catch (Exception ex)
            {
                return new LoginResponse
                {
                    IsSuccess = false,
                    Message = "Erro na autenticação"
                };
            }
        }

        public async Task<List<string>> GetUserPermissionsAsync(string username)
        {
            try
            {
                // TODO: Integrar com servicoV.getpermissoes(username)
                await Task.Delay(50);
                
                // Simulação - retorna permissões baseadas no usuário
                var permissions = new List<string>();
                
                // Simula diferentes níveis de acesso
                if (username.ToLower().Contains("admin") || username == "alexander.valerio")
                {
                    permissions.Add("@@@@"); // Acesso total
                }
                else
                {
                    permissions.AddRange(new[]
                    {
                        "PAINEL TI",
                        "PAINEL RH",
                        "PAINEL FINANCEIRO"
                    });
                }
                
                return permissions;
            }
            catch
            {
                return new List<string>();
            }
        }
    }
}